src/preload.py

```diff
@@ -846,7 +886,7 @@ def get_docker_storage_driver(image=None):
     assert False, "Docker storage driver could not be found"
 
 
-def main_preload(app_data, additional_bytes):
+def main_preload(app_data, additional_bytes, balenaOS_version=""):
```

Comment:
There is no need for a default value.

Reply:
Instead of passing the version here, you could pass the splash image destination (`SPLASH_IMAGE_TO`), that way you could use the already installed `semver` on the node side and not have to add it in `requirements.txt`.